axelrod/tests/strategies/test_meta.py

```diff
@@ -59,6 +60,36 @@ def test_repr(self):
             ),
         )
 
+    def test_clone(self):
+        # Test that the cloned player produces identical play
+        player1 = self.player()
+        if player1.name in ["Darwin", "Human"]:
+            # Known exceptions
+            return
+        player2 = player1.clone()
+        self.assertEqual(len(player2.history), 0)
+        self.assertEqual(player2.cooperations, 0)
+        self.assertEqual(player2.defections, 0)
+        self.assertEqual(player2.state_distribution, {})
+        self.assertEqual(player2.classifier, player1.classifier)
+        self.assertEqual(player2.match_attributes, player1.match_attributes)
+
+        turns = 10
+        for op in [
+            axelrod.Cooperator(),
+            axelrod.Defector(),
+            axelrod.TitForTat(),
+        ]:
+            player1.reset()
+            player2.reset()
+            seed = random.randint(0, 10 ** 6)
```

Comment:
This could prove tricky to pin down in case of a failure (which is unlikely). Could you perhaps wrap this as a hypothesis test (with the seed being randomly picked by hypothesis)?

Reply:
For this I copied the test from [TestPlayer](https://github.com/Axelrod-Python/Axelrod/blob/583cc4df87d6ea5f0eb0a08000e4b8b26bae56d3/axelrod/tests/strategies/test_player.py#L447), lowered the number of rounds, and took out the stochastic opponent. I changed the seeding on that test as well.